metricflow/test/integration/test_cases/itest_metrics.yaml

```diff
@@ -1194,6 +1194,26 @@ integration_test:
     GROUP BY
       1
 ---
+integration_test:
+  name: simple_fill_nulls_with_0_agg_time_dim
+  description: Test a simple query that joins to time spine and fills nulls, with agg_time_dimension
+  model: SIMPLE_MODEL
+  metrics: ["bookings_fill_nulls_with_0"]
+  group_by_objs: [{"name": "booking__ds__day"}]
+  check_query: |
+    SELECT
+      subq_5.ds AS booking__ds__day
+      , COALESCE(subq_3.bookings, 0) AS bookings_fill_nulls_with_0
+    FROM {{ source_schema }}.mf_time_spine subq_5
+    LEFT OUTER JOIN (
```

Comment:
Ok, so previously this would have just run the query but skipped the join, correct? That seems strange and incorrect to me.

Reply:
The logic there was that if there's no time dimension to join to, we can't join to time spine anyway.
And then, if there is a time dimension that isn't agg time, should we time spine join to that dimension? Seems inconsistent if we're treating it like any other categorical dimension in all other ways.